gpustat.py

```diff
@@ -198,139 +192,110 @@ def uuid(self):
-        smi_output = execute_process(
-            r'nvidia-smi --query-compute-apps={query_cols} --format=csv,noheader,nounits'.format(
-                query_cols=','.join(gpu_query_columns)
-            ))
-
-        process_entries = []
-        for line in smi_output.split('\n'):
-            if not line: continue
-            query_results = line.split(',')
-            process_entry = dict({col_name: col_value.strip() for
-                                  (col_name, col_value) in zip(gpu_query_columns, query_results)
-                                  })
-            process_entries.append(process_entry)
-
-        pid_map = {int(e['pid']) : None for e in process_entries
-                   if not 'Not Supported' in e['pid']}
-
-        # 2. map pid to username, etc.
-        if pid_map:
-            # Sometimes nvidia-smi returns non-existent process PID (see #12);
-            # To let ps exit with a non-zero return code in such cases,
-            # we always include querying PID 1 as well (but ignored).
-            pid_output = execute_process('ps -o {} -p1 -p {}'.format(
-                'pid,user:16,comm',
-                ','.join(map(str, pid_map.keys()))
-            ))
-
-            for line in pid_output.split('\n'):
-                if (not line) or 'PID' in line: continue
-                pid, user, comm = line.split()[:3]
-
-                pid = int(pid)
-                if pid <= 1: continue
-
-                pid_map[pid] = {
-                    'user' : user,
-                    'comm' : comm
-                }
-
-        # 3. add some process information to each process_entry
-        for process_entry in process_entries[:]:
-
-            if 'Not Supported' in process_entry['pid']:
-                # TODO move this stuff into somewhere appropriate
-                # such as running_processes(): process_entry = ...
-                # or introduce Process class to elegantly handle it
-                process_entry['user'] = None
-                process_entry['comm'] = None
-                process_entry['pid'] = None
-                process_entry['used_memory'] = None
-                continue
-
-            pid = int(process_entry['pid'])
-
-            if pid_map[pid] is None:
-                # !?!? this pid is listed up in nvidia-smi's query result,
-                # but actually seems not to be a valid running process. ignore!
-                process_entries.remove(process_entry)
-                continue
-
-            process_entry.update(pid_map[pid])
-
-        return process_entries
-
-    def update_process_information(self):
-        processes = self.running_processes()
-        for p in processes:
+            name = _decode(N.nvmlDeviceGetName(handle))
+            uuid = _decode(N.nvmlDeviceGetUUID(handle))
+
+            try:
+                temperature = N.nvmlDeviceGetTemperature(handle, N.NVML_TEMPERATURE_GPU)
+            except N.NVMLError:
+                memory = None  # Not supported
```

Comment:
It looks like a typo. Maybe `temperature = None` should be in except?

Reply:
Absolutely, you are correct. Fixed in 895e1f8. Thanks!